src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/kafka/OppfolgingstilfelleConsumer.kt

```diff
@@ -0,0 +1,97 @@
+package no.nav.syfo.oppfolgingstilfelle.kafka
+
+import no.nav.syfo.personstatus.application.OppfolgingstilfelleService
+import no.nav.syfo.personstatus.infrastructure.database.DatabaseInterface
+import no.nav.syfo.personstatus.infrastructure.kafka.KafkaConsumerService
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.ConsumerRecords
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+class OppfolgingstilfelleConsumer(
+    val database: DatabaseInterface,
+    val oppfolgingstilfelleService: OppfolgingstilfelleService,
+) : KafkaConsumerService<OppfolgingstilfellePersonRecord> {
+
+    override val pollDurationInMillis: Long = 1000
+
+    override fun pollAndProcessRecords(kafkaConsumer: KafkaConsumer<String, OppfolgingstilfellePersonRecord>) {
+        val records = kafkaConsumer.poll(Duration.ofMillis(pollDurationInMillis))
+        if (records.count() > 0) {
+            processRecords(records)
+            kafkaConsumer.commitSync()
+        }
+    }
+
+    private fun processRecords(
+        records: ConsumerRecords<String, OppfolgingstilfellePersonRecord>,
+    ) {
+        val (tombstoneRecordList, recordsValid) = records.partition { it.value() == null }
+
+        processTombstoneRecordList(tombstoneRecordList = tombstoneRecordList)
+
+        val recordsRelevant = recordsValid
+            // Hvorfor trenger vi å sortere?
+            .sortedWith(
+                compareByDescending<ConsumerRecord<String, OppfolgingstilfellePersonRecord>> { record ->
+                    record.value().referanseTilfelleBitInntruffet
+                }.thenByDescending { it.value().createdAt }
```

Comment:
Hvorfor sorterer vi her? 🤔 Vi går vel gjennom alle uansett, men rekkefølgen har kanskje noe å si?

Reply:
Vi ønsker vel å prosessere sånn at de blir kronologisk riktig rekkefølge (så hvis det finnes to varianter av det samme oppfølgingstilfellet er det det nyeste som blir prosessert til slutt). Tror dette har vært riktig fra før.